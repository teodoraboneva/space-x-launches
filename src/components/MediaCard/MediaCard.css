.media-card .MuiCardMedia-root {
  height: 0;
  padding-top: 100%;
  margin: auto;
}
