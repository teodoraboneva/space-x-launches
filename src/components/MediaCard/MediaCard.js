import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";

import "./MediaCard.css";

const MediaCard = ({ image, heading, children }) => (
  <Card className="media-card">
    <CardActionArea>
      <CardMedia image={image} title="" />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {heading}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="div">
          {children}
        </Typography>
      </CardContent>
    </CardActionArea>
  </Card>
);

export default MediaCard;
