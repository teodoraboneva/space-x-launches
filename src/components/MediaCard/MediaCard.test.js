import renderer from "react-test-renderer";
import MediaCard from "./MediaCard";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MediaCard image="image" heading="FalconSat #1">
        <p>Launch Year: 2006</p>
        <p>Successful Launch: false</p>
      </MediaCard>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
