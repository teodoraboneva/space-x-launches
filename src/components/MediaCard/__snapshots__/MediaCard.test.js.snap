// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root media-card MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root"
      style={
        Object {
          "backgroundImage": "url(\\"image\\")",
        }
      }
      title=""
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        FalconSat #1
      </h2>
      <div
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        <p>
          Launch Year: 2006
        </p>
        <p>
          Successful Launch: false
        </p>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
